### ISSUE TITLE
Bug: Envoy failing on its own with a critical error

### ISSUE THREAD

**Issue by pkrawiec8** [aws/aws-app-mesh-roadmap#130]
**Summary**
From time to time one of the ECS tasks is failing due to the Envoy container.
We are using the newest Envoy image: `840364872350.dkr.ecr.eu-west-1.amazonaws.com/aws-appmesh-envoy:v1.11.2.0-prod`

Error is: `Stopped reason: Essential container in task exited`

**Steps to Reproduce**
Not able to easily reproduce.

**Are you currently working around this issue?**
Currently we have no workarounds.

**Additional context**
Possibly related to: https://github.com/envoyproxy/envoy/issues/6557
The fix is on the master branch on Envoy, but not on `v1.11.2`

**Attachments**
Stack trace:
```
[2019-10-30 09:05:27.925][1][warning][config] [bazel-out/k8-fastbuild/bin/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 0,
[2019-10-30 09:36:14.295][1][warning][config] [bazel-out/k8-fastbuild/bin/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 0,
[2019-10-30 10:08:08.263][1][critical][assert] [source/common/upstream/health_checker_impl.cc:176] assert failure: !response_headers_.
[2019-10-30 10:08:08.263][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Aborted, suspect faulting address 0x53900000001
[2019-10-30 10:08:08.263][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-10-30 10:08:08.285][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: Envoy::SignalAction::sigHandler() [0x1a0333e]
[2019-10-30 10:08:08.285][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: __restore_rt [0x7f09460061f0]
[2019-10-30 10:08:08.307][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: Envoy::Http::StreamDecoderWrapper::decodeHeaders() [0x121d8ef]
[2019-10-30 10:08:08.328][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: Envoy::Http::Http1::ClientConnectionImpl::onHeadersComplete() [0x13bd889]
[2019-10-30 10:08:08.402][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: Envoy::Http::Http1::ConnectionImpl::onHeadersCompleteBase() [0x13bb3b9]
[2019-10-30 10:08:08.423][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: Envoy::Http::Http1::ConnectionImpl::{lambda()#5}::operator()() [0x13b99fe]
[2019-10-30 10:08:08.497][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: Envoy::Http::Http1::ConnectionImpl::{lambda()#5}::_FUN() [0x13b9a1d]
[2019-10-30 10:08:08.519][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: http_parser_execute [0x1669c70]
[2019-10-30 10:08:08.593][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Http::Http1::ConnectionImpl::dispatchSlice() [0x13bab47]
[2019-10-30 10:08:08.615][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Http::Http1::ConnectionImpl::dispatch() [0x13ba9e7]
[2019-10-30 10:08:08.636][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: Envoy::Http::CodecClient::onData() [0x12ca4e4]
[2019-10-30 10:08:08.706][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #11: Envoy::Http::CodecClient::CodecReadFilter::onData() [0x12cb028]
[2019-10-30 10:08:08.727][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #12: Envoy::Network::FilterManagerImpl::onContinueReading() [0xfa95d0]
[2019-10-30 10:08:08.801][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #13: Envoy::Network::FilterManagerImpl::onRead() [0xfa9750]
[2019-10-30 10:08:08.822][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #14: Envoy::Network::ConnectionImpl::onRead() [0xf9f31c]
[2019-10-30 10:08:08.896][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #15: Envoy::Network::ConnectionImpl::onReadReady() [0xfa0b87]
[2019-10-30 10:08:08.917][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #16: Envoy::Network::ConnectionImpl::onFileEvent() [0xfa0925]
[2019-10-30 10:08:08.938][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #17: Envoy::Network::ConnectionImpl::ConnectionImpl()::{lambda()#3}::operator()() [0xf9cead]
[2019-10-30 10:08:09.012][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #18: std::_Function_handler<>::_M_invoke() [0xfa32a9]
[2019-10-30 10:08:09.033][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #19: std::function<>::operator()() [0xf96948]
[2019-10-30 10:08:09.106][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #20: Envoy::Event::FileEventImpl::assignEvents()::{lambda()#1}::operator()() [0xf965a4]
[2019-10-30 10:08:09.127][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #21: Envoy::Event::FileEventImpl::assignEvents()::{lambda()#1}::_FUN() [0xf96622]
[2019-10-30 10:08:09.201][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #22: event_process_active_single_queue.isra.33 [0x16403b9]
[2019-10-30 10:08:09.222][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #23: event_base_loop [0x164399f]
[2019-10-30 10:08:09.296][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #24: Envoy::Event::LibeventScheduler::run() [0xfb9018]
[2019-10-30 10:08:09.317][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #25: Envoy::Event::DispatcherImpl::run() [0xf9191b]
[2019-10-30 10:08:09.338][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #26: Envoy::Server::InstanceImpl::run() [0xeef319]
[2019-10-30 10:08:09.411][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #27: Envoy::MainCommonBase::run() [0x43d7ca]
[2019-10-30 10:08:09.432][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #28: Envoy::MainCommon::run() [0x420fa4]
[2019-10-30 10:08:09.504][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #29: main [0x41ed91]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #30: __libc_start_main [0x7f094591402a]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Segmentation fault, suspect faulting address 0x0
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:69] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: Envoy::SignalAction::sigHandler() [0x1a0333e]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: __restore_rt [0x7f09460061f0]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: Envoy::Http::StreamDecoderWrapper::decodeHeaders() [0x121d8ef]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: Envoy::Http::Http1::ClientConnectionImpl::onHeadersComplete() [0x13bd889]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: Envoy::Http::Http1::ConnectionImpl::onHeadersCompleteBase() [0x13bb3b9]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #5: Envoy::Http::Http1::ConnectionImpl::{lambda()#5}::operator()() [0x13b99fe]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #6: Envoy::Http::Http1::ConnectionImpl::{lambda()#5}::_FUN() [0x13b9a1d]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #7: http_parser_execute [0x1669c70]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #8: Envoy::Http::Http1::ConnectionImpl::dispatchSlice() [0x13bab47]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #9: Envoy::Http::Http1::ConnectionImpl::dispatch() [0x13ba9e7]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #10: Envoy::Http::CodecClient::onData() [0x12ca4e4]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #11: Envoy::Http::CodecClient::CodecReadFilter::onData() [0x12cb028]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #12: Envoy::Network::FilterManagerImpl::onContinueReading() [0xfa95d0]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #13: Envoy::Network::FilterManagerImpl::onRead() [0xfa9750]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #14: Envoy::Network::ConnectionImpl::onRead() [0xf9f31c]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #15: Envoy::Network::ConnectionImpl::onReadReady() [0xfa0b87]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #16: Envoy::Network::ConnectionImpl::onFileEvent() [0xfa0925]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #17: Envoy::Network::ConnectionImpl::ConnectionImpl()::{lambda()#3}::operator()() [0xf9cead]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #18: std::_Function_handler<>::_M_invoke() [0xfa32a9]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #19: std::function<>::operator()() [0xf96948]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #20: Envoy::Event::FileEventImpl::assignEvents()::{lambda()#1}::operator()() [0xf965a4]
[2019-10-30 10:08:09.506][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #21: Envoy::Event::FileEventImpl::assignEvents()::{lambda()#1}::_FUN() [0xf96622]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #22: event_process_active_single_queue.isra.33 [0x16403b9]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #23: event_base_loop [0x164399f]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #24: Envoy::Event::LibeventScheduler::run() [0xfb9018]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #25: Envoy::Event::DispatcherImpl::run() [0xf9191b]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #26: Envoy::Server::InstanceImpl::run() [0xeef319]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #27: Envoy::MainCommonBase::run() [0x43d7ca]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #28: Envoy::MainCommon::run() [0x420fa4]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #29: main [0x41ed91]
[2019-10-30 10:08:09.507][1][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #30: __libc_start_main [0x7f094591402a]
```

**Comment by lavignes** [comment]
Since Envoy 1.12.0 has been released (and contains the fix) and this error seems relatively infrequent. I think we should favor releasing 1.12.0 as the fix for this issue: https://github.com/aws/aws-app-mesh-roadmap/issues/132

**Comment by lavignes** [comment]
We have now released the updated image with the fix for this issue: #132 

Feel free to re-open if you see something similar again :)

**Comment by pkrawiec8** [comment]
Thanks for updating it :) Hope it will resolve the problem.
